crates/arroyo-server-common/src/tls.rs

```diff
@@ -1,25 +1,82 @@
-use anyhow::{anyhow, Result};
-use arroyo_rpc::config::TlsConfig;
+use anyhow::{anyhow, bail, Result};
+use arroyo_rpc::config::{config, ApiAuthMode, TlsConfig};
 use arroyo_rpc::native_cert_store;
 use axum_server::tls_rustls::RustlsConfig;
+use rustls::pki_types::pem::PemObject;
+use rustls::pki_types::{CertificateDer, PrivateKeyDer};
+use rustls::server::WebPkiClientVerifier;
 use rustls::{ClientConfig, RootCertStore, ServerConfig};
 use std::sync::Arc;
-use tonic::transport::{Identity, ServerTlsConfig};
+use tonic::transport::{Certificate, Identity, ServerTlsConfig};
 
 /// Create TLS configuration for Axum HTTP/HTTPS servers
-pub async fn create_http_tls_config(tls_config: &TlsConfig) -> Result<RustlsConfig> {
+pub async fn create_http_tls_config(
```

Comment:
Maybe this should be renamed?  Not really just the tls config anymore.  More of auth config and configuring TLS is one of the auth strategies.

Reply:
This method is still just handling TLS config (possibly mTLS). HTTP auth is handled elsewhere, when creating the server.